gc.c

```diff
@@ -15101,9 +15103,35 @@ static size_t rb_mmtk_available_system_memory(void)
     return rb_mmtk_system_physical_memory();
 }
 
+static size_t
+rb_mmtk_parse_heap_limit(char *argv)
+{
+    int pow = 0;
+    size_t len = strlen(argv);
+    char last = argv[len - 1];
+
+    // shorthand for TB, MB and GB. eg. 1T, 1G, 1M etc
+    switch(last) {
+        case 'T': 
+            pow = 40; // gigabytes. 2^30
+            break;
+        case 'G': 
+            pow = 30; // gigabytes. 2^30
+            break;
+        case 'M': 
+            pow = 20; // mebibytes. 2^20
+            break;
+    }
+    argv[len - 1] = '\0';
```

Comment:
Doesn't this erase the last digit in the case where there is no size modifier?

Reply:
Good spot, thank you. Fixed.